app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/MainNavGraph.kt

```diff
@@ -0,0 +1,263 @@
+        vm.homeEvent.LaunchCollection { navController.navigateAsTopLevel(Routes.Home) }
+    }
+}
+
+private fun ComposeActivity.devOptionsComposable(navGraphBuilder: NavGraphBuilder, navController: NavController) {
+    navGraphBuilder.composable(Routes.DevOptions) {
+        val viewModel: DevOptionsViewModel = getViewModel()
+        viewModel.ConnectBaseViewModel {
+            DevOptionsScreen(state = it.toState())
+        }
+
+        // Only debug/internal builds allowed to show this screen. Immediately close if somehow launched on prod release build.
+        if (!activityViewModel.devOptionsEnabled) {
+            navController.navigateUp()
+        }
+
+        controls.title = stringResource(id = R.string.dev_options_title)
+    }
+}
+
+private fun ComposeActivity.homeComposable(navGraphBuilder: NavGraphBuilder, navController: NavController) {
+    navGraphBuilder.composable(Routes.Home) {
+        val viewModel: HomeViewModel = getViewModel()
+
+        viewModel.ConnectBaseViewModel {
+            HomeScreen(state = it.toState())
+        }
+
+        controls.title = stringResource(id = R.string.home_title)
+        controls.topLevel = true
+
+        viewModel.itemSelected.LaunchCollection{
+            activityViewModel.selectedRepo.value = it.repo
+            navController.navigate(Routes.RepositoryBrowser(
+                RepositoryBrowserData(repoName = it.repo.name ?: "")
+            ))
+        }
+    }
+}
+
+private fun ComposeActivity.repositoryFileComposable(navGraphBuilder: NavGraphBuilder) {
+    navGraphBuilder.composable(
+        "file?hash={hash}&path={path}&mimeType={mimeType}",
+        arguments = listOf(
+            navArgument("hash") {},
+            navArgument("path") {},
+            navArgument("mimeType") { nullable = true },
+        )
+    ) { backStackEntry ->
+        val data = RepositoryFileData(
+            hash = backStackEntry.arguments?.getString("hash") ?: "",
+            path = backStackEntry.arguments?.getString("path") ?: "",
+            mimeType = backStackEntry.arguments?.getString("mimeType") ?: "",
+        )
+
+        val viewModel: RepositoryFileFragmentViewModel = getViewModel()
+        viewModel.ConnectBaseViewModel {
+            FileBrowserScreen(state = it.toState())
+            activityViewModel.selectedRepo.LaunchCollection { repo ->
+                it.loadFile(
+                    repo.workspace?.slug ?: "",
+                    repo.name ?: "",
+                    data.mimeType,
+                    data.hash,
+                    data.path
+                )
+            }
+        }
+
+        controls.title = data.path
+    }
+}
+
+private fun ComposeActivity.repositoryBrowserComposable(navGraphBuilder: NavGraphBuilder, navController: NavController) {
+    navGraphBuilder.composable(
+        "repository?repoName={repoName}&folderHash={folderHash}&folderPath={folderPath}",
```

Comment:
@BottleRocket-Colin nit: Nice to have the named arg (`route`) used here

```suggestion
        route = "repository?repoName={repoName}&folderHash={folderHash}&folderPath={folderPath}",
```